userprofile/templates/userprofile/profile.html

```diff
@@ -9,89 +9,114 @@
-					<input name="next" type="hidden" value="{{ redirect_to }}">
-					<div class="input-group">
-						<select name="language" class="form-control">
-							{% get_current_language as LANGUAGE_CODE %}
-							{% get_available_languages as LANGUAGES %}
-							{% get_language_info_list for LANGUAGES as languages %}
-							{% for language in languages %}
-								<option value="{{ language.code }}" {% if language.code == LANGUAGE_CODE %}selected{% endif %}>
-									{{ language.name_local }}
-								</option>
-							{% endfor %}
-						</select>
-						<div class="input-group-btn">
-							<button type="submit" class="btn btn-primary">{% trans 'Save' %}</button>
-						</div>
-					</div>
-				</form>
-			</dd>
-        </dl>
-    </div>
-    <div class="col-md-6">
-        <div class="alert alert-info">
-            {% blocktrans with email=user.email %}
-            You can change your avatar icon
-            at <a href="http://gravatar.com">gravatar.com</a>.
-            Use your e-mail address <strong>{{ email }}</strong>
-            when registering your account.
-            {% endblocktrans %}
-        </div>
-
-        {# TODO: check the user type and show editing options/help #}
-        <div class="alert alert-info">
-            {% blocktrans %}
-            To change your settings, such as <strong>e-mail</strong>
-            address or <strong>password</strong>, you must update
-            them on your identity provider.
-            {% endblocktrans %}
-        </div>
+	<div class="col-sm-offset-2 col-sm-8 col-lg-4">
+		<h2>{% trans "A+ Preferences:" %}</h2>
+		<label for="api-access-token">{% trans "API Access Token" %}</label>
+		<div class="input-group">
+			<input readonly type="text" class="form-control read-only" id="api-access-token"
+				value="{{ profile.api_token }}">
+			<span class="input-group-btn">
+				<button data-toggle="tooltip" data-placement="bottom" data-copy="{{ profile.api_token }}"
+					title="{% trans 'Copy API Access Token!' %}" aria-label="Copy API Access Token"
+					class="aplus-button--secondary aplus-button--md js-copy" type="button"><span
+						class="glyphicon glyphicon-copy"></span><span class="">{% trans 'Copy' %}</span></button>
+			</span>
+		</div>
+		<br>
+		<div>
+			<label for="local-storage-services">{% trans "Automatically log in to:" %}</label>
+			<ul id="local-storage-services" class="local-storage-fields"
+				data-forget-text="{% trans "Forget automatic redirection" %}"></ul>
+		</div>
 		<div class="well">
-			<h4>{% trans "Automatic service redirections" %}</h4>
 			<p>
 				{% blocktrans %}
 				These are all services that you have marked for automatic redirection on this device.
-				You can remove automatic redirection here and you will be shown the privacy information next time you are being redirected to the service.
+				You can remove automatic redirection here and you will be shown the privacy information next time you
```

Comment:
Should this text about automatic redirection be placed before the list of the services? Users now see the list before the instructions explaining the list. Additionally, if there are no services, then these instructions don't connect to anything. Can you add text like "no services marked" if the list is empty?